Add ComfyUI Asset Downloader

mythz opened the pull request.

mythz committed:
Add ComfyUI Asset Downloader

ltdrdata commented:
https://github.com/ServiceStack/comfy-asset-downloader/blob/78bee0e18db8917c5ea0ebddb592e13d3caa5472/nodes/asset_downloader.py#L68

The `save_to` path must be sanitized. Allowing write access to locations via relative paths such as `..` could enable downloading arbitrary remote files to any location, which constitutes a critical security vulnerability.

And it is also recommended to access the model path through the `folder_paths` module in the core.

`folder_paths.get_folder_paths()`, `folder_paths.models_dir`

mythz commented:
Thanks for the feedback! I've changed it to resolve `models_dir` from `folder_paths` and sanitized `save_to` to ensure its within the models_dir in this commit: https://github.com/ServiceStack/comfy-asset-downloader/commit/81ff3d8a6b7dee5ea4ae564c9367ad70350d3771

I wasn't able to get the list of directories and immediate sub directories from `get_folder_paths()` so I've just changed it to look at  `folder_paths.models_dir`
